cmd/notary-server/config.go

```diff
@@ -256,3 +260,34 @@ func parseServerConfig(configFilePath string, hRegister healthRegister) (context
 		ConsistentCacheControlConfig: consistentCache,
 	}, nil
 }
+
+func setupConfigReloadTrap(oldConfig *viper.Viper, configFile string, envPrefix string) {
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, notary.NotarySupportedSignals...)
+	go func() {
+		for {
+			reloadConfig(<-c, configFile, oldConfig, envPrefix)
+		}
+	}()
+}
+
+func reloadConfig(sig os.Signal, configFile string, oldConfig *viper.Viper, envPrefix string) {
+	reload := func(config *viper.Viper) {
+		switch sig {
+		case syscall.SIGUSR1:
+			if err := utils.AdjustLogLevel(true); err != nil {
+				fmt.Println("Increase log level failed, will use the old one, error: ", err)
```

Comment:
Why using `fmt.Print` not `logrus` here and below is because it will not log out anything if the log level is too low, `Panic` for example and we can't just use `logrus.Fatalf()` for it will terminate the whole process.